react/week4/gitusers/src/components/Users.js

```diff
@@ -0,0 +1,61 @@
+import React, {useState, useEffect, createContext} from 'react'
+import SearchUser from './SearchUser'
+import {FaSearch} from 'react-icons/fa'
+const UserContext = createContext()
+
+
+const Users = () => {
+    const [users, setUsers] = useState([])
+    const [loading, setLoading] = useState(true)
+    const [error, setError] = useState(null)
+    
+   const fetchGithubApi = async(inputValue)=>{
+    const API_URL = `https://api.github.com/search/users?q=${inputValue}+in:user`
+
+        try{
+            setLoading(true)
+            const response = await fetch(API_URL)
+            if(!response.ok){
+                throw Error('No data could be fetched from the url')
+            }
+
+            const githubData = await response.json()
+            const githubUsers = githubData.items
+            setUsers(githubUsers)
+            setLoading(false)
+          }
+        catch(error){
+            setError(`Error fetching: ${error.message}`)
+            setLoading(false)
+        }
+        
+    }
+ 
+    useEffect(()=>{
+        fetchGithubApi(); 
+    }, [])
+
```

Comment:
```suggestion
    useEffect(()=>{
        fetchGithubApi(textInput); 
    }, [textInput])

```

Reply:
Maybe fetchGithubUsers is better name for the function 🙂

Reply:
Thnakyou so much for the feedback